src/javaServerStarter.ts

```diff
@@ -78,6 +102,47 @@ function prepareParams(requirements: RequirementsData, xmlJavaExtensions: string
   return params;
 }
 
+async function getServerBinary(): Promise<string> {
+  let server_home: string = path.resolve(__dirname, '../server');
+  let binaries: Array<string> = glob.sync(`**/lemminx-${os.platform()}*`, { cwd: server_home });
+  const JAR_REJECTOR: RegExp = /\.jar$/;
+  binaries = binaries.filter((path) => { return !JAR_REJECTOR.test(path) });
+  if (binaries.length) {
+    return new Promise<string>((resolve, reject) => {
+      resolve(path.resolve(server_home, binaries[0]));
+    });
+  }
+  // Download it, then return the downloaded binary's location
+  return downloadBinary();
+}
+
+/**
+ * Downloads lemminx binary under the `server` directory and returns the path to the binary as a Promise
+ *
+ * @returns The path to the binary as a Promise
+ */
+async function downloadBinary(): Promise<string> {
+  console.log('Downloading lemminx binary server...');
```

Comment:
need to download the hash too

Reply:
we might need to declare hash/binary url pairs in package.json eventually (at build time)

Reply:
That was my plan; package the hashes of a given version of the server in the release of the .vsix